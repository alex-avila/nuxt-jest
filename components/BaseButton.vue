<template>
  <button
    :class="withoutStyle ? 'has-no-style' : 'button is-primary'"
    v-bind="$attrs"
    v-on="$listeners"
  >
    <slot />
  </button>
</template>

<script>
export default {
  props: {
    withoutStyle: {
      type: Boolean,
      default: false
    }
  }
}
</script>


<style lang="sass" scoped>
.has-no-style
  color: inherit
  background: none
  border: none
  font: inherit
  cursor: inherit
</style>
