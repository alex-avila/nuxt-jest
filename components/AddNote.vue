<template>
  <div class="add-note">
    <div
      ref="editor"
      :class="['box', {'has-text-weight-bold': note.length}]"
      contenteditable="true"
      v-bind="$listeners"
      @input="$emit('handleUpdate', $event)"
      @focus="togglePlaceholder(false)"
      @blur="togglePlaceholder(true)"
    >
      {{ placeholder }}
    </div>
    <div class="button__wrapper">
      <base-button
        :disabled="!note.length"
        @click="$emit('addNote', $refs.editor)"
      >
        Start a New Note
      </base-button>
    </div>
  </div>
</template>

<script>
import BaseButton from './BaseButton.vue'

export default {
  components: {
    BaseButton
  },

  props: {
    note: {
      type: String,
      required: true
    }
  },

  data: () => ({ placeholder: 'Type to enter a new note...' }),

  methods: {
    togglePlaceholder(bool) {
      if (bool && !this.note.length) {
        this.$refs.editor.innerHTML = this.placeholder
      } else if (!bool && !this.note.length) {
        this.$refs.editor.innerHTML = ''
      }
    }
  }
}
</script>

<style lang="sass" scoped>
.button__wrapper
  display: flex
  justify-content: center

.add-note
  margin: auto
  max-width: 500px
  width: 100%
  text-align: left
</style>
