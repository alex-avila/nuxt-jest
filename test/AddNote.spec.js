import { shallowMount } from '@vue/test-utils'
import AddNote from '@/components/AddNote.vue'
import BaseButton from '@/components/BaseButton.vue'

describe('AddNote', () => {
  const wrapper = shallowMount(AddNote, {
    propsData: {
      note: ''
    }
  })
  const editableDiv = wrapper.find('[contenteditable="true"]')
  const button = wrapper.find(BaseButton)

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  test('has default placeholer', () => {
    expect(editableDiv.element.innerHTML.trim()).toBe(
      wrapper.vm.$data.placeholder
    )
  })

  test('button is initially disabled', () => {
    expect(button.attributes('disabled')).toBeTruthy()
  })
})
