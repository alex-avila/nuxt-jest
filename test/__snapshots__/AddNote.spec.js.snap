// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddNote renders correctly 1`] = `
<div
  class="add-note"
>
  <div
    class="box"
    contenteditable="true"
  >
    
    Type to enter a new note...
  
  </div>
   
  <div
    class="button__wrapper"
  >
    <base-button-stub
      disabled="true"
    >
      
      Add Note
    
    </base-button-stub>
  </div>
</div>
`;
